services/headless-lms/src/models/pages.rs

```diff
@@ -1182,70 +1181,74 @@ pub async fn get_page_search_results_for_phrase(
+    let last_word_option = page_search_request
         .query
         .trim()
         .split_ascii_whitespace()
-        .last()
-        // If not found we'll skip the replacement with a random word not found in the search corpus
-        .unwrap_or("randomuniquewordforskippingthereplacement332");
+        .last();
 
-    let res = sqlx::query_as!(
-        PageSearchResult,
-        "
+    let res = if let Some(last_word) = last_word_option {
+        sqlx::query_as!(
+            PageSearchResult,
+            "
 -- common table expression for the search term tsquery so that we don't have to repeat it many times
 WITH cte as (
-  -- Converts the search term to a phrase search with phraseto_tsquery but appends ':*' to the last word so that it
-  -- becomes a prefix match. This way the search will also contain results when the last word in the search term
-  -- is only partially typed. Note that if to_tsquery($4) decides to stem the word, the replacement will be skipped.
-  SELECT ts_rewrite(
-      phraseto_tsquery($2::regconfig, $3),
-      to_tsquery($4),
-      to_tsquery($4 || ':*')
+    -- Converts the search term to a phrase search with phraseto_tsquery but appends ':*' to the last word so that it
+    -- becomes a prefix match. This way the search will also contain results when the last word in the search term
+    -- is only partially typed. Note that if to_tsquery($4) decides to stem the word, the replacement will be skipped.
+    SELECT ts_rewrite(
+        phraseto_tsquery($2::regconfig, $3),
+        to_tsquery($4),
+        to_tsquery($4 || ':*')
     ) as query
 )
 SELECT id,
-  ts_rank(
+    ts_rank(
     content_search,
     (
-      SELECT query
-      from cte
+        SELECT query
+        from cte
     )
-  ) as rank,
-  ts_headline(
+    ) as rank,
+    ts_headline(
     $2::regconfig,
     title,
     (
-      SELECT query
-      from cte
+        SELECT query
+        from cte
     )
-  ) as title_headline,
-  ts_headline(
+    ) as title_headline,
+    ts_headline(
     $2::regconfig,
     content_search_original_text,
     (
-      SELECT query
-      from cte
+        SELECT query
+        from cte
     )
-  ) as content_headline,
-  url_path
+    ) as content_headline,
+    url_path
 FROM pages
 WHERE course_id = $1
-  AND deleted_at IS NULL
-  AND content_search @@ (
+    AND deleted_at IS NULL
+    AND content_search @@ (
     SELECT query
     from cte
-  )
+    )
 ORDER BY rank DESC
 LIMIT 50;
-    ",
-        course_id,
-        course.content_search_language as _,
-        page_search_request.query,
-        last_word
-    )
-    .fetch_all(conn)
-    .await?;
+        ",
+            course_id,
+            course.content_search_language as _,
+            page_search_request.query,
+            last_word
+        )
+        .fetch_all(conn)
+        .await?
+    } else {
+        // If last word not found the search query is empty and we can return an empty vec
+        Vec::new()
```

Comment:
I think it would be more clear if the None case was handled as early as possible to avoid large conditional blocks, e.g.
```rust
    let last = if let Some(last) = page_search_request
        .query
        .trim()
        .split_ascii_whitespace()
        .last()
    {
        last
    } else {
        return Vec::new();
    };
```
Not a big deal though